pkg/agent/config/traffic_encap_mode_test.go

```diff
@@ -0,0 +1,248 @@
+package config
+
+import (
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetTrafficEncapModeFromStr(t *testing.T) {
+	tests := []struct {
+		name    string
+		str     string
+		expBool bool
+		expMode TrafficEncapModeType
+	}{
+		{
+			name:    "encap-mode-valid",
+			str:     "enCap",
+			expBool: true,
+			expMode: 0,
+		},
+		{
+			name:    "no-encap-mode-valid",
+			str:     "Noencap",
+			expBool: true,
+			expMode: 1,
+		},
+		{
+			name:    "hybrid-mode-valid",
+			str:     "Hybrid",
+			expBool: true,
+			expMode: 2,
+		},
+		{
+			name:    "invalid-str",
+			str:     "en cap",
+			expBool: false,
+			expMode: -1,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
for _, tt := range tests {
                t.Run(tt.name, func(t *testing.T) {
                     actualBool,......
                      ....
                })